### PR TITLE
sys-libs/libcxx: fix build on clang macos prefix

### PR DESCRIPTION
**Pull request by DarthGandalf**
Bug: https://bugs.gentoo.org/758167

---

Please check all the boxes that apply:

- [ ] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [ ] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [ ] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [ ] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: sys-libs/libcxx/files/libcxx-19.1.3-macos-reexports.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Fix build of standalone libc++ on Gentoo Prefix
+
+Inspired by:
+https://github.com/llvm/llvm-project/pull/79012
+https://github.com/llvm/llvm-project/pull/110920
```
Comment (thesamesam) [inline sys-libs/libcxx/files/libcxx-19.1.3-macos-reexports.patch:5]: Is this upstream somewhere or does it need to be sent? If the latter, please do so.

Comment (DarthGandalf) [reply sys-libs/libcxx/files/libcxx-19.1.3-macos-reexports.patch:5]: Click the links :) One of them is how it broke, another one is the fix. But the fix is more involved than this patch, as it's doing more stuff. I'm here partially reverting the breaking change.

**Comment by DarthGandalf** [comment]
I updated only this version, because 20.0.0_pre20241029 is already fixed upstream.

File: sys-libs/libcxx/files/libcxx-19.1.3-macos-reexports.patch (hunk 1 above)
Comment (thesamesam) [reply sys-libs/libcxx/files/libcxx-19.1.3-macos-reexports.patch:5]: I did look! But it wasn't clear to me if those were related (hence "inspired") or just a similar type of problem. Please explicitly add that it's essentially a partial revert and that it's fixed on master.

Comment (DarthGandalf) [reply sys-libs/libcxx/files/libcxx-19.1.3-macos-reexports.patch]: Done

**Review by thesamesam — APPROVED** [review]
LGTM, not sure if @mgorny wants to add it to patchset if it's only for 19.

**Comment by mgorny** [comment]
Was the fix queued for 19.x backport upstream? I'd rather see it fixed there while they still accept fixes.

**Comment by DarthGandalf** [comment]
> Was the fix queued for 19.x backport upstream?

As this was fixed upstream already, I didn't think about such backport upstream

**Comment by mgorny** [comment]
Could you confirm that https://github.com/llvm/llvm-project/pull/115892 helps for you?

**Comment by DarthGandalf** [comment]
It removes the `target_link_libraries(cxx_shared PRIVATE cxxabi-reexports)` line, which was the issue, so looks like yes.

**Comment by mgorny** [comment]
Could you actually test it, please? To make sure that's 100% sufficient and doesn't introduce any regressions.

**Comment by DarthGandalf** [comment]
> Could you actually test it, please? 

Done. It builds if I replace my patch with this patch, on top of the same snapshot of 19.

> 100% sufficient and doesn't introduce any regressions.

The regressions which I was worrying about (which is why I simplified the patch so much) was other uses, e.g. building both libcxxabi and libcxx together

**Comment by thesamesam** [comment]
I think this patch is in 19.1.4.

**Pull request closed without merging**